src/Extensions/Abstractions/ExecutionContext/BaseExecutionContext.cs

```diff
@@ -52,6 +55,22 @@ public BaseExecutionContext(IHostEnvironment? hostEnvironment = null)
 			}
 
 			IsCanary = false;
+
+			// Service Fabric specific values
+			ServiceName = GetVariable(ServiceNameVariableName) ?? DefaultEmptyValue;
+			ApplicationName = GetVariable(ApplicationNameVariableName) ?? DefaultEmptyValue;
+
+			string nodeName = GetVariable(NodeNameVariableName) ?? DefaultEmptyValue;
+			MachineId = FormattableString.Invariant($"{MachineName}_{nodeName}");
+
+			string? nodeIPAddressOrFQDN = GetVariable(NodeIPOrFQDNVariableName);
+
+			if (IPAddress.TryParse(nodeIPAddressOrFQDN, out IPAddress? ipAddress))
+			{
+				ClusterIpAddress = ipAddress;
```

Comment:
How is machine IP becoming cluster IP? Cluster can have multiple machines.

Reply:
Also a copy of existing behavior in our current infrastructure, I'm not attached to it if you have suggestions about new approach